addons/website_slides_survey/controllers/slides.py

```diff
@@ -23,29 +23,38 @@ def _get_slide_detail(self, slide):
-        result = super(WebsiteSlides, self)._get_slide_detail(slide)
+        result = super(WebsiteSlides, self)._get_slide_detail(slide, is_fullscreen)
         if not request.env.user._is_public() and slide.slide_type == 'certification' and slide.survey_id:
-            result['certification_test_entry'] = not slide.channel_id.is_member
-            result['certification_done'] = False
-            if slide.channel_id.is_member:
-                user_membership_id_sudo = slide.user_membership_id.sudo()
-                quizz_passed = user_membership_id_sudo.survey_quizz_passed
-                result['certification_done'] = quizz_passed
-                result['survey_id'] = slide.survey_id.id
-                if not quizz_passed:
-                    last_user_input = next(user_input for user_input in user_membership_id_sudo.user_input_ids.sorted(
-                        lambda user_input: user_input.create_date, reverse=True
-                    ))
-                    result['certification_url'] = last_user_input._get_survey_url()
+            if not is_fullscreen:
+                result['certification_test_entry'] = not slide.channel_id.is_member
+                result['certification_done'] = False
+                if slide.channel_id.is_member:
+                    user_membership_id_sudo = slide.user_membership_id.sudo()
+                    quizz_passed = user_membership_id_sudo.survey_quizz_passed
+                    result['certification_done'] = quizz_passed
+                    result['survey_id'] = slide.survey_id.id
+                    if not quizz_passed:
+                        last_user_input = next(user_input for user_input in user_membership_id_sudo.user_input_ids.sorted(
+                            lambda user_input: user_input.create_date, reverse=True
+                        ))
+                        result['certification_url'] = last_user_input._get_survey_url()
+                else:
+                    user_input = slide.survey_id._create_answer(
+                        partner=request.env.user.partner_id,
+                        check_attempts=False,
+                        test_entry=True, **{
+                            'slide_id': slide.id
+                        }
+                    )
+                    result['certification_url'] = user_input._get_survey_url()
             else:
-                user_input = slide.survey_id._create_answer(
-                    partner=request.env.user.partner_id,
-                    check_attempts=False,
-                    test_entry=True, **{
-                        'slide_id': slide.id
+                result['certifications_infos'] = {
+                        s['id']: {
+                            'certification_url': s['certification_url'],
+                            'certification_done': s['certification_done'],
+                            'survey_id': s['survey_id']
+                        } for s in slide.action_get_slide_survey_urls()
```

Comment:
As far as I know this method is only called with a browse record of only one slide (hence the name of the variable "slide" instead of "slides").

I don't really get the thinking behind this method.
Maybe @jem-odoo has more insight on this?